File: software/examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_moisture import Moisture

# Callback for moisture value greater than 200
def cb_reached(moisture):
    print('Moisture Value: ' + str(moisture))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    m = Moisture(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 1 seconds (1000ms)
    m.set_debounce_period(1000)

    # Register threshold reached callback to function cb_reached
    m.register_callback(m.CALLBACK_MOISTURE_REACHED, cb_reached)

    # Configure threshold for "greater than 200"
    m.set_moisture_callback_threshold('>', 200, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/php/ExampleThreshold.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletMoisture.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletMoisture;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

// Callback function for moisture value reached callback
function cb_moistureReached($moisture)
{
    echo "Moisture Value: $moisture\n";
}

$ipcon = new IPConnection(); // Create IP connection
$m = new BrickletMoisture(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get threshold callbacks with a debounce time of 1 second (1000ms)
$m->setDebouncePeriod(1000);

// Register moisture value reached callback to function cb_moistureReached
$m->registerCallback(BrickletMoisture::CALLBACK_MOISTURE_REACHED, 'cb_moistureReached');

// Configure threshold for moisture value "greater than 200"
$m->setMoistureCallbackThreshold('>', 200, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
